### PR TITLE
feat: Support phone number during egagement contact creation & retrieval

### PR DESCRIPTION
**Pull request by tonyxiao**
## Test Plan

Manually tested

- [x] Get contact with phone in salesloft (we had a bug, this PR fixed it)
- [x] Create contact with phone in Salesloft
- [x] Create contact with phone in Outreach
- [x] Create contact with phone in Apollo (not officially supported by Apollo in docs, but works emprically)
- [x] Update contact with phone in Apollo

## Future

We should add better typing to `record` and perhaps even runtime validation so this type of error wouldn't occur without being caught.

### CODE REVIEW

**Review by lucasmarshall — APPROVED** [review]

File: packages/core/remotes/impl/salesloft/mappers.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ export const fromSalesloftPersonToContact = (record: Record<string, any>): Conta
       postalCode: null,
     },
     emailAddresses: fromSalesloftPersonToEmailAddresses(record),
-    phoneNumbers: fromSalesloftPersonToPhoneNumbers(record.phone_numbers ?? []),
+    phoneNumbers: fromSalesloftPersonToPhoneNumbers(record),
```
Comment (asdfryan) [inline packages/core/remotes/impl/salesloft/mappers.ts:44]: Thanks for the fix! Could you add / modify an existing unit test? (in `salesloft/mappers.test.ts`)

File: packages/core/remotes/impl/outreach/mappers.ts [hunk 2]
```diff
@@ -503,6 +503,10 @@ export const toOutreachProspectPhoneNumbers = (phoneNumbers?: PhoneNumber[]) =>
         case 'work':
           workPhones.push(phoneNumber);
           break;
+        case 'primary':
+          // Defaulting to work phone for primary
```
Comment (asdfryan) [inline packages/core/remotes/impl/outreach/mappers.ts:507]: `s/work/mobile` ?

Also can you add a test case for this? (and the case where we have both primary + mobile)

Comment (tonyxiao) [reply packages/core/remotes/impl/outreach/mappers.ts:507]: 👍

File: packages/core/remotes/impl/apollo/mappers.ts [hunk 3]
```diff
@@ -190,6 +193,14 @@ export const toApolloContactCreateParams = (params: ContactCreateParams): Record
     email: params.emailAddresses?.[0]?.emailAddress,
     present_raw_address: params.address ? getRawAddressString(params.address) : undefined,
     account_id: params.accountId,
+    // Apollo docs appears to not support phone during creation
+    // However empirically it actually works, so we keep the phone mapping logic here.
```
Comment (tomkit) [inline packages/core/remotes/impl/apollo/mappers.ts:197]: 🙏 for the comments here

**Pull request merged**